api/src/lib.rs

```diff
@@ -2589,4 +2594,1264 @@ pub extern "system" fn Java_com_horizen_sigproofnative_NaiveThresholdSigProof_na
+    _cert_verification_key: jbyteArray,
+    _btr_verification_key_nullable: jbyteArray, // can be null if there is no key for BTRs
+    _csw_verification_key_nullable: jbyteArray, // can be null if there is no key for CSWs
+    _tx_hash: jbyteArray,
+    _out_idx: jint
+) -> jboolean
+{
+    let sc_id = {
+        let t = _env.convert_byte_array(_sc_id)
+            .expect("Should be able to convert to Rust array");
+
+        let mut sc_id_bytes = [0u8; FIELD_SIZE];
+
+        t.write(&mut sc_id_bytes[..])
+            .expect("Should be able to write into byte array of fixed size");
+
+        sc_id_bytes
+    };
+
+    let amount = _amount as i64;
+
+    let pub_key = {
+        let t = _env.convert_byte_array(_pub_key)
+            .expect("Should be able to convert to Rust array");
+
+        let mut pub_key_bytes = [0u8; 32];
+
+        t.write(&mut pub_key_bytes[..])
+            .expect("Should be able to write into byte array of fixed size");
+
+        pub_key_bytes
+    };
+
+    let withdrawal_epoch_length = _withdrawal_epoch_length as u32;
+
+    let custom_data = _env.convert_byte_array(_custom_data)
+        .expect("Should be able to convert to Rust byte array");
+
+    let mut _constant_nullable_vec;
+    let constant = if _constant_nullable.is_null() {
+        Option::None
+    } else {
+        _constant_nullable_vec = _env.convert_byte_array(_constant_nullable)
+            .expect("Should be able to convert to Rust byte array");
+        Some(_constant_nullable_vec.as_slice())
+    };
+
+    let cert_verification_key = _env.convert_byte_array(_cert_verification_key)
+        .expect("Should be able to convert to Rust byte array");
+
+    let mut _btr_verification_key_nullable_vec;
+    let btr_verification_key = if _btr_verification_key_nullable.is_null() {
+        Option::None
+    } else {
+        _btr_verification_key_nullable_vec = _env.convert_byte_array(_btr_verification_key_nullable)
+            .expect("Should be able to convert to Rust byte array");
+        Some(_btr_verification_key_nullable_vec.as_slice())
+    };
+
+    let mut _csw_verification_key_nullable_vec;
+    let csw_verification_key = if _csw_verification_key_nullable.is_null() {
+        Option::None
+    } else {
+        _csw_verification_key_nullable_vec = _env.convert_byte_array(_csw_verification_key_nullable)
+            .expect("Should be able to convert to Rust byte array");
+        Some(_csw_verification_key_nullable_vec.as_slice())
+    };
+
+    let tx_hash = {
+        let t = _env.convert_byte_array(_tx_hash)
+            .expect("Should be able to convert to Rust array");
+
+        let mut tx_hash_bytes = [0u8; 32];
+
+        t.write(&mut tx_hash_bytes[..])
+            .expect("Should be able to write into byte array of fixed size");
+
+        tx_hash_bytes
+    };
+
+    let out_idx = _out_idx as u32;
+
+    let commitment_tree = {
+
+        let t =_env.get_field(_commitment_tree, "commitmentTreePointer", "J")
```

Comment:
to avoid duplications of such a part of the code the method like this could be useful:
https://github.com/nastenko/zendoo-sc-cryptolib/blob/76e533cd3ff2de91871f354f599633140644c03e/api/src/zenbox_smt/jni_calls.rs#L31